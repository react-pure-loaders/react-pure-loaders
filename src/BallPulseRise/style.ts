import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';

const riseAmount = '30px';

const ballPulseRise = keyframes({
  '0%': {
    transform: 'scale(1.1)',
  },
  '100%': {
    transform: 'scale(1.0)',
  },
  '25%': {
    transform: `translateY(-${riseAmount})`,
  },
  '50%': {
    transform: 'scale(0.4)',
  },
  '75%': {
    transform: `translateY(${riseAmount})`,
  },
  '99%': {
    transform: 'translateY(0)',
  },
});

export default (color?: string) => ({
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    animationDelay: '0',
    animationDuration: '1s',
    animationIterationCount: 'infinite',
    animationTimingFunction: 'cubic-bezier(.15,.46,.9,.6)',
    display: 'inline-block',

    '&:nth-of-type(2n)': {
      animationName: ballPulseRise,
    },

    '&:nth-of-type(2n-1)': {
      animationName: ballPulseRise,
    },
  },
});
