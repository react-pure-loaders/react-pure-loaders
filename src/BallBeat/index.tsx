/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
