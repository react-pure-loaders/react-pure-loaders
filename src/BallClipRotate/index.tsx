/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
