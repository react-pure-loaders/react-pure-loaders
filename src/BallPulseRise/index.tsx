/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
