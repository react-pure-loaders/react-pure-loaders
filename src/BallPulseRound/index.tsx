/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
