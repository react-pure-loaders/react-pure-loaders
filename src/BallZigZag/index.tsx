/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
