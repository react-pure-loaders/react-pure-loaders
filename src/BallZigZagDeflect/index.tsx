/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
