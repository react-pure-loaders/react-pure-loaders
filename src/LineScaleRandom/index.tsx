/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
