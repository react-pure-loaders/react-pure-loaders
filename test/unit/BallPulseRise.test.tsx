import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallPulseRise from '../../src/BallPulseRise';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallPulseRise>', () => {
    afterEach(cleanup);

    test('BallPulseRise should match snapshot', () => {
        const { container } = render(<BallPulseRise loading={true} />);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('BallPulseRise should have default color', () => {
        const { container } = render(<BallPulseRise loading={true} />);

        expect(container.firstChild).toHaveStyleRule(
            'background-color',
            PRIMARY_COLOR,
            { target: '> div' },
        );
    });

    test.skip('BallPulseRise should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(
            <BallPulseRise color={color} loading={true} />,
        );

        expect(container.firstChild).toHaveStyleRule('background-color', color, {
            target: '> div',
        });
    });

    test('BallPulseRise should have no children', () => {
        const { container } = render(<BallPulseRise loading={false} />);

        expect(container.firstChild).toBeNull();
    });

    test('BallPulseRise should have three children', () => {
        const { container } = render(<BallPulseRise loading={true} />);

        expect(container.querySelectorAll('div')).toHaveLength(6);
    });
});
