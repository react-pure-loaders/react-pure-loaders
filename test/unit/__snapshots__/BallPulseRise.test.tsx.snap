// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallPulseRise> BallPulseRise should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scale(1.1);
    -moz-transform: scale(1.1);
    -ms-transform: scale(1.1);
    transform: scale(1.1);
  }

  100% {
    -webkit-transform: scale(1.0);
    -moz-transform: scale(1.0);
    -ms-transform: scale(1.0);
    transform: scale(1.0);
  }

  25% {
    -webkit-transform: translateY(-30px);
    -moz-transform: translateY(-30px);
    -ms-transform: translateY(-30px);
    transform: translateY(-30px);
  }

  50% {
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }

  75% {
    -webkit-transform: translateY(30px);
    -moz-transform: translateY(30px);
    -ms-transform: translateY(30px);
    transform: translateY(30px);
  }

  99% {
    -webkit-transform: translateY(0);
    -moz-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scale(1.1);
    -moz-transform: scale(1.1);
    -ms-transform: scale(1.1);
    transform: scale(1.1);
  }

  100% {
    -webkit-transform: scale(1.0);
    -moz-transform: scale(1.0);
    -ms-transform: scale(1.0);
    transform: scale(1.0);
  }

  25% {
    -webkit-transform: translateY(-30px);
    -moz-transform: translateY(-30px);
    -ms-transform: translateY(-30px);
    transform: translateY(-30px);
  }

  50% {
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }

  75% {
    -webkit-transform: translateY(30px);
    -moz-transform: translateY(30px);
    -ms-transform: translateY(30px);
    transform: translateY(30px);
  }

  99% {
    -webkit-transform: translateY(0);
    -moz-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }
}

.emotion-0>div {
  border-radius: 100%;
  height: 15px;
  margin: 2px;
  width: 15px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: cubic-bezier(.15,.46,.9,.6);
  animation-timing-function: cubic-bezier(.15,.46,.9,.6);
  display: inline-block;
}

.emotion-0>div:nth-of-type(2n) {
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
}

.emotion-0>div:nth-of-type(2n-1) {
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
